src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -216,6 +225,8 @@ internal class ConnectionManager : IConnectionManager {
         }
 
         private Dictionary<string, IConnection> CreateConnectionList() {
+            _shell.AssertIsOnMainThread();
```

Comment:
`CreateConnectionList` is called from `ConnectionManager` constructor which is accessed from `RInteractiveWorkflow` constructor. There is no guarantee that it is called from UI thread, so this assert will eventually fail.